ports/esp32/README.md

```diff
@@ -26,59 +26,122 @@ There are two main components that are needed to build the firmware:
   different to the compiler used by the ESP8266)
 - the Espressif IDF (IoT development framework, aka SDK)
 
-The ESP-IDF changes quickly and MicroPython only supports certain versions. The
-git hash of these versions (one for 3.x, one for 4.x) can be found by running
-`make` without a configured `ESPIDF`. Then you can fetch only the given esp-idf
-using the following command:
-
-    $ git clone https://github.com/espressif/esp-idf.git
-    $ git checkout <Current supported ESP-IDF commit hash>
-    $ git submodule update --init --recursive
-
-Note: The ESP IDF v4.x support is currently experimental.
-
-The binary toolchain (binutils, gcc, etc.) can be installed using the following
-guides:
-
-  * [Linux installation](https://docs.espressif.com/projects/esp-idf/en/stable/get-started/linux-setup.html)
-  * [MacOS installation](https://docs.espressif.com/projects/esp-idf/en/stable/get-started/macos-setup.html)
-  * [Windows installation](https://docs.espressif.com/projects/esp-idf/en/stable/get-started/windows-setup.html)
-
-If you are on a Windows machine then the
-[Windows Subsystem for Linux](https://msdn.microsoft.com/en-au/commandline/wsl/install_guide)
-is the most efficient way to install the ESP32 toolchain and build the project.
-If you use WSL then follow the
-[Linux guidelines](https://esp-idf.readthedocs.io/en/latest/get-started/linux-setup.html)
-for the ESP-IDF instead of the Windows ones.
-
-You will also need either Python 2 or Python 3, along with the `pyserial` and
-`pyparsing` packages installed for the version of Python that you will be using
-(when building you can use, eg, `make PYTHON=python2` to specify the version
-used).  To install the required packages do:
+If you are on a Windows machine then the [Windows Subsystem for
+Linux](https://msdn.microsoft.com/en-au/commandline/wsl/install_guide) is the
+most efficient way to install the ESP32 toolchain and build the project. If
+you use WSL then follow the Linux instructions rather than the Windows
+instructions.
+
+The ESP-IDF changes quickly and MicroPython only supports certain versions.
+The git hash of these versions (one for 3.x, one for 4.x) can be found by
+running `make` without a configured `ESPIDF`. Then you can fetch the
+required IDF using the following command:
+
+```bash
+$ cd ports/esp32
+$ make ESPIDF=
+$ cd $HOME/src/github.com/espressif  # Or any path you like
```

Comment:
Maybe start with `setenv EDPIDF=$HOME/src/github.com/espressif  # Or any path you like` here and then you can use that env variable in the further explanations below?